framework/doc/content/syntax/FunctorMaterials/index.md

```diff
@@ -0,0 +1,169 @@
+# FunctorMaterials
+
+Functor material properties are properties that are evaluated on-the-fly.
+Along with numerous classes of MOOSE objects, they are [Functors](syntax/Functors/index.md). This allows
```

Comment:
```suggestion
Along with multiple other classes of MOOSE objects, they are [Functors](syntax/Functors/index.md). This allows
```
Only three main systems are functors at the current time, so numerous feels like a bit too much. Maybe not forever though!